Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -316,6 +319,89 @@ WarpX::MacroscopicEvolveE (int lev, PatchType patch_type, amrex::Real a_dt) {
     }
 }
 
+void
+WarpX::DampFieldsInGuards(std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield,
+                          std::array<std::unique_ptr<amrex::MultiFab>,3>& Bfield) {
+
+    for ( amrex::MFIter mfi(*Efield[0], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi )
```

Comment:
Question: Do we lose generality by choosing to use, here and in the following, the component `0` of the `Efield` MultiFab?

Reply:
Good question. For the MFIter object, it won't matter since all of the fields have the same blocking structure. For the tileboxes, yes it could matter. With RZ though, all of the fields have the same centering so have the same tileboxes. For Cartesian, the fields could be staggered so the different fields could have difference tileboxes. For the most general case, a ParallelFor would be written for each field component. It's complicated since the different staggerings would have different z locations - would they need different damping factors? @RemiLehe do you think that this would be needed?